src/components/Dropdown/Dropdown.tsx

```diff
@@ -143,107 +148,144 @@ const Dropdown = React.forwardRef<HTMLDivElement, DropdownProps>(
   ) => {
     const Comp = asChild ? Slot : 'div';
 
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
     const [isOpen, setIsOpen] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
     const [selectedId, setSelectedId] = useState<string>();
     const [selectedValue, setSelectedValue] = useState<OptionsType | undefined>(
       undefined,
     );
-    const selectableOptions = options.filter((opt) => !opt.category);
 
-    useMemo(() => {
-      if (value) {
-        setSelectedValue(value);
-        setSelectedId(value.id);
-      }
-    }, [value]);
+    const selectableOptions = useMemo(() => {
+      return options.filter((opt) => !opt.category);
+    }, [options]);
 
-    const handleChange = (value: OptionsType) => {
-      const isSelected = selectedId === value.id;
-      setSelectedId(isSelected ? undefined : value.id);
-      setSelectedValue(isSelected ? undefined : value);
+    const switchDropDownOpenState = useCallback(() => {
```

Comment:
Bad name, sounds like it toggles state but it always closes the dropdown. Perhaps "closeDropdown" would be better?

Reply:
I agree. This is changed to "closeDropdown"